docs/guides/code-style-guide.md

```diff
@@ -0,0 +1,161 @@
+
+As you write code, you may make changes to the variables that you are using. You may rename some variables, make some new variables or change how the variables are used. This may leave some variables that are *unused*.
+
+```js
+function orderTaxi(pickUpTime) {
+  let driverName = getDriverName();
+  let customerName = getCustomerName(); // 🛑 Don't do this!
+
+  return `${driverName} will pick you up at ${pickUpTime}`;
+}
+```
+
+In this example, the `customerName` variable isn't used anywhere.
+
+You should remove any variables that are unused. This is because if you (or someone else) is reading your code, it can be confusing if you see a variable and then find out later that it isn't used. It could make you think that there's a bug, because the variable must have been put there for a reason!
+
+## Think of good names for your variables
+
+Making our code understandable by others is critical to being a programmer. One of the main tools in our toolbox is good naming for our variables and function names.
+
+There are no strict rules to follow when thinking of variable names, but there are some general guidelines. 
+
+:::tip 
+When thinking about whether a variable name is good, try to imagine that you are reading the code again in the future and you have forgotten exactly how it works.
+
+Do the variable names help explain what the code is supposed to do?
+:::
+
+### Use `camelCase`
+
+When writing JavaScript, programmers tend to use a convention called *`camelCase`*. This means starting your variable name with a lower case letter, and then every "word" after that starts with an upper case letter.
+
+- `startingLocation`
+- `timeToDestination`
+- `trafficOnRoute`
+
+This convention makes it easier to read variables that have multiple "words".
+
+### Avoid short names
+
+Very short variable names can be difficult to understand since the purpose of the variable can be unclear. They are also difficult remember, especially if there are many similarly named variables. Try to avoid short names or abbreviations.
+
+Here are some examples of **bad** names that you should avoid:
+
+- Single letters like `x` or `y`
+- Abbreviations like `evt` instead of `event`
+
+### Describe what the variable is/does
+
+A good variable name quickly explains what it represents to anyone reading the code. Try to describe what the variable is or what it does within in code.
+
+```js
+// 🛑 Try to avoid this
+let song = true;
+
+// ✅ This is (probably) better
+let isPlaying = true;
+```
+
+In this example, the variable name `song` doesn't tell us what it is used for. The `isPlaying` variable name is better since it tells us  whether a song is playing or not.
+
+```js
+// 🛑 Try to avoid this
+function percentage() {
+  // ...
+}
+
+// ✅ This is (probably) better
+function getPercentage() {
+  // ...
+}
+```
+
+In this example, the function calculates a percentage and returns it. Therefore it is good to name it using the "get" verb to show that it returns something.
+
```

Comment:
```suggestion
In this example, the function calculates a percentage and returns it. Therefore it is good to name it using the "get" verb to show that it returns something.

```js
// 🛑 Try to avoid this
function isOldEnough(number) {
  // ...
}

// ✅ This is (probably) better
function isOldEnough(yearOfBirth) {
  // ...
}
```

In this example, we expect the caller of the function to pass a year of birth. The name "number" isn't very clear about what number should be passed. The name "yearOfBirth" makes it more clear, so is a better name. Some functions _are_ very generic, e.g. a function which reverses an array without caring what's stored in it may call its parameter `array`, but try to think of the most specific name that makes sense.

```